libraries/abstractions/pkcs11/psa/iot_pkcs11_psa_object_management.c

```diff
@@ -0,0 +1,600 @@
+/*
+ * Copyright (C) 2017 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ * Copyright (c) 2019 Arm Limited. All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/*
+ * This file is derivative of amazon-freertos\libraries\3rdparty\mbedtls\library\
+ * pkparse.c(amazon-freertos commit 74875b1d2)
+ */
+
+/**
+ * @file iot_pkcs11_psa_object_management.c
+ * @brief Functions of managing the objects stored in TF-M.
+ */
+
+#include <string.h>
+#include "iot_pkcs11_psa_object_management.h"
+#include "iot_pkcs11_psa_input_format.h"
+
+P11KeyConfig_t P11KeyConfig __attribute__(( section( "tasks_share" ) )) =
+{.uxDeviceCertificate = 5U,
+ .uxJitpCertificate = 6U,
+ .uxRootCertificate = 7U,
+};
+
+/**
+* @brief Writes a file to local storage.
+*
+* Port-specific file write for crytographic information.
+*
+* @param[in] pxClass	   Class of the object to be saved.
+* @param[in] pxLabel	   Label of the object to be saved.
+* @param[in] pucData	   Data buffer to be written to file
+* @param[in] ulDataSize	Size (in bytes) of data to be saved.
+* @param[in] pvContext	 Context of the key.
+*
+* @return The file handle of the object that was stored.
+*/
+CK_OBJECT_HANDLE PKCS11_PSA_SaveObject( CK_ATTRIBUTE_PTR pxClass,
+	CK_ATTRIBUTE_PTR pxLabel,
+	uint8_t * pucData,
+	uint32_t ulDataSize,
+	mbedtls_pk_context *pvContext )
+{
+	CK_OBJECT_HANDLE xHandle = eInvalidHandle;
+	psa_key_handle_t uxKeyHandle;
+	CK_ULONG uxKeyType;
+	psa_key_policy_t policy = psa_key_policy_init();
+	psa_algorithm_t uxAlgorithm;
+	CK_RV xBytesWritten = 0;
+	CK_RV xReturn;
+	uint32_t ulPresentMark = ( pkcs11OBJECT_PRESENT_MAGIC | ( ulDataSize ) );
+	uint8_t * pemBuffer;
+	size_t pemLength;
+	const psa_ps_create_flags_t uxFlags = PSA_PS_FLAG_NONE;
+	psa_ps_status_t uxStatus;
+	psa_ecc_curve_t curve_id;
+	const mbedtls_ecp_keypair *ec;
+	unsigned char *pcPrivateKeyDerStart = NULL;
```

Comment:
define buffer for raw private key (see comment at line 198
        unsigned char pcPrivateKeyRaw[MBEDTLS_ECP_MAX_BYTES];
        size_t xPrivateKeySizeRaw = 0;

Reply:
See the reply at line 198.